### PR TITLE
Check for tunnel failures from browserstack

### PR DESCRIPTION
**Pull request by ankoor7**

### CODE REVIEW

**Comment by ankoor7** [comment]
@rgamm could i get a quick PR?

**Comment by timsweetman** [comment]
```
} else if (err.code === 13) {
    options.log('Selenium timeout: ' + task.descriptor + ' - adding to the back of the queue.');
    task.seleniumTimeout();
    if (task.waitTimeouts === 1) {
        // todo add config option
        silentRequeue = true;
    }
} else if (err.code === 255) {
    options.log('BrowserStack Tunnel failure: ' + task.descriptor + ' -  adding to the back of the queue ' + err + stdout);
    task.seleniumTimeout();
    silentRequeue = true;
```
- Do you really get code 255 only when a browerstack tunnel failed, or do random other failures also give 255?
- What does silentRequeue do, and why are we ignoring it in this case?
- The Travis checks didn't work. I don't know why.

Could this be refactored?

```
if (errorIndicatesSeleniumFailure(err)) {
    options.log("Selenium failure - adding to back of the queue...")
```

(where errorIndicatesSeleniumFailure knows how to interpret err)

**Comment by mike-suggitt** [comment]
Why don't you scrap the lot and go with a more standardised, well-supported lib such as webdriverio? :)

**Comment by timsweetman** [comment]
Does that help? These are browserstack failures.

**Comment by mike-suggitt** [comment]
Not neccessarily but as a library it has a far better support of browserstack. As this file change is inside grunt/gulp task it would unlikely have any direct impact but it seems v odd that the ssh tunnel would fail

**Comment by timsweetman** [comment]
> v odd that the ssh tunnel would fail

Intermittent failures, in a test suite? How unusual. #sarcasm

(Of course, it's possible that something dreadful is happening)

**Comment by mike-suggitt** [comment]
lol in the tests yea def but not in an established ssh tunnel - unless app gets re-built mid-way through. Anyways yea you should invest in webdriverio/cucumberjs framework in Node :)

**Comment by timsweetman** [comment]
> you should invest in webdriverio/cucumberjs framework in Node

[No, we shouldn't](http://www.joelonsoftware.com/articles/fog0000000069.html). :)

**Comment by mike-suggitt** [comment]
lol classic BBC :P

**Comment by ankoor7** [comment]
@timsweetman to answer your questions:
1) Do you really get code 255 only when a browerstack tunnel failed, or do random other failures also give 255?
- Maybe, but you get a 255 when an ssh connection fails. So, in general contexts, this could be from authentication, a host not being in known_hosts when it should be, or the tunnel not existing.
  Should i limit the number of retries here or make a different condition?

2) What does silentRequeue do, and why are we ignoring it in this case?
- It determines if a task is re-queued without affecting the retry count. Here we want it to be restarted without incrementing the retry count because the test hasn't started yet.

3) The Travis checks didn't work. I don't know why.
- This is down to the version of node used in that travis check, so, nor do I as yet. Note there are 2 travis jobs with different versions of node and npm. The build fails with npm version 1.2.30 and not the later 2.15.1.
  The errors come from npm 1.2.30, on node 0.8, being unable to resolve suitable versions of modules.

**Comment by ankoor7** [comment]
We now use the latest version of node and npm. 
It may be a good idea to use the latest version grunt-parallel-behat. We forked a very old version and kept using it

**Comment by ankoor7** [comment]
We use the version of node and npm that is on the passing travis job

**Comment by ankoor7** [comment]
@mike-suggitt The problem is that the ssh tunnel is not established

> lol in the tests yea def but not in an established ssh tunnel - unless app gets re-built mid-way through. Anyways yea you should invest in webdriverio/cucumberjs framework in Node :)

File: npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+File exists: /usr/local/bin/grunt
+Move it away, and try again.
+94 error System Darwin 15.5.0
+95 error command "/usr/local/bin/node" "/usr/local/bin/npm" "install" "-g" "grunt" "grunt-cli"
+96 error cwd /Users/patela33/workspace/grunt-parallel-behat
+97 error node -v v0.8.17
+98 error npm -v 1.2.0
+99 error path /usr/local/bin/grunt
+100 error code EEXIST
+101 verbose exit [ 1, true ]
```
Comment (timsweetman) [inline npm-debug.log]: You accidentally a log file

Comment (ankoor7) [reply npm-debug.log]: #git fail. Thanks for resolving that

**Pull request merged**